### PR TITLE
WC2-603_WC2-604: ETL debug page not showing first visit or shows exit date for ben not yet exited

### PR DESCRIPTION
**Pull request by butofleury**
Explain what problem this PR is resolving

Related JIRA tickets : [WC2-603](https://bluesquare.atlassian.net/browse/WC2-603), [WC2-604](https://bluesquare.atlassian.net/browse/WC2-604)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

Fixed ETL to store both admission and followup visits linked to beneficiary.  The bug was mainly appearing for Nigeria data.

## How to test

From the local instance with wfp coda database, run ETL script to generate data for:
- Nigeria:  `docker compose run iaso manage etl_ng`
- South Sudan: `docker compose run iaso manage etl_ssd`

Then, try to access to debug page for beneficiary belongs to account:

- Nigeria: [http://localhost:8081/wfp/debug/684/](http://localhost:8081/wfp/debug/684/)
- South Sudan: [http://localhost:8081/wfp/debug/323/](http://localhost:8081/wfp/debug/323/)

You should see all visits linked to beneficiary and correct exit date when beneficiary has discharged.

## Print screen / video

**1. Nigeria**

image

**2. South Sudan**

image

[WC2-603]: https://bluesquare.atlassian.net/browse/WC2-603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WC2-604]: https://bluesquare.atlassian.net/browse/WC2-604?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: plugins/wfp/management/commands/south_sudan/Under5.py [hunk 1]
```diff
@@ -163,20 +166,20 @@ def run(self):
...
-
-                    followUpVisits = ETL().group_followup_steps(grouped_steps, admission_step)
-                    steps = ETL().save_steps(visits, followUpVisits)
-                    logger.info(f"Inserted {len(steps)} Steps")
-                else:
-                    logger.info("No new journey")
-            logger.info(
-                f"---------------------------------------------------------------------------------------------\n\n"
-            )
+            if beneficiary is not None:
```
Comment (bramj) [inline plugins/wfp/management/commands/south_sudan/Under5.py:169]: Should there be some error notification or something if `beneficiary` is `None`?

**Review by bramj — APPROVED** [review]
I don't know the exact details of the logic, but looks okay to me :+1:

File: plugins/wfp/management/commands/south_sudan/Under5.py (hunk 1 above)
Comment (butofleury) [reply plugins/wfp/management/commands/south_sudan/Under5.py:169]: @bramj , yep, on QA, there a some strange data with beneficiary with medical visit without anthropometric during admission visit.It has happened for Children under 5 only for South Sudan and it was violating some DB constraint.

**Comment by beygorghor** [comment]
@bramj @butofleury is there a way to test this on dev or on QA please ?

**Comment by butofleury** [comment]
> @bramj @butofleury is there a way to test this on dev or on QA please ?

@beygorghor , to test this on [QA](https://qa.coda.go.wfp.org/), you should login with 2 different user, 1 linked to Nigeria and another linked to South Sudan. Then, open debug page for 1 beneficiray with more than 1 visit.

e.g:

-  Nigeria : debug page for [beneficiary](https://qa.coda.go.wfp.org/wfp/debug/666/ ),
-  South Sudan, for [beneficiary](https://qa.coda.go.wfp.org/wfp/debug/323/)

You should see more than 1 visit(2 visits for Nigeria and 3 for South Sudan) for each beneficiary.

**Comment by beygorghor** [comment]
@butofleury @bramj  , both debug pages are accessible while connected to a South Sudan account, is this the desired effect ?

**Pull request merged**